### PR TITLE
update bioconductor packages

### PR DESCRIPTION
**Pull request by pabloaledo**
This PR updates several bioconductor packages to use more standard names, and also use commit-id's instead of version numbers for better reproducibility.

### CODE REVIEW

**Comment by marcodelapierre** [comment]
@alalazo  FYI

**Review by alalazo — DISMISSED** [review]
Looks good to me. Let me check how to deal with the name changes. I assume these packages are widely used, so I want to double check if we need anything more than have an entry in the CHANGELOG for the v0.22 release.

**Comment by marcodelapierre** [comment]
@alalazo  please let me know if I can be of any assistance on this last leg of work before the merge

**Comment by marcodelapierre** [comment]
@pabloaledo this should have come to my mind earlier. See this old PR of mine where I wanted to rename e.g. `tower-cli` to `nf-tower-cli`: https://github.com/spack/spack/pull/34515/files

Instead of renaming, I was asked to duplicate the package recipes, and `deprecate` the old ones. Those will then be removed in a subsequent release of Spack.

@alalazo  I think this would address your concern, right?

**Comment by alalazo** [comment]
@marcodelapierre Yes, that works for sure. Apologies for not getting back earlier, but tried to contact previous maintainer without much success.

**Comment by marcodelapierre** [comment]
No worries, thank you Max!

**Comment by marcodelapierre** [comment]
@pabloaledo  please add deprecation attributes as per: https://github.com/spack/spack/pull/34515/files#diff-6c3a337d07dc30fd0ede96af6f5cf4d84df89afc0886807aea5abea1f7819c59

on the actual removal: just noticed that my packages are still there one year later, opening a PR for removal now :-)

**Comment by marcodelapierre** [comment]
@alalazo  we have re-added the old named recipes (`bioconductor-` instead of `r-`), with the `deprecate` attribute. 

Once the latest `develop` is merged in (@pabloaledo ),  it hould be safe to merge into Develop!

File: var/spack/repos/builtin/packages/r-ebseq/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/r-ebseq/package.py:9]: If i read the diff correctly, this one was not restored and deprecated.

Comment (pabloaledo) [reply var/spack/repos/builtin/packages/r-ebseq/package.py:9]: Sorry!, forgot that one :facepalm:

Comment (alalazo) [reply var/spack/repos/builtin/packages/r-ebseq/package.py:9]: No worries, thanks!

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks!

**Comment by marcodelapierre** [comment]
Thank you Max!

**Pull request merged**
